adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -530,6 +668,61 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Prepares two RDDs to be joined with any shuffleRegionJoin. This includes copartition
+   * and sort of the rightRdd if necessary.
+   *
+   *  @param genomicRdd The RDD to join to.
+   *  @param optPartitions An optional number of partitions.
+   *  @return a case class containing all the prepared data for ShuffleRegionJoins
+   */
+  private def prepareForShuffleRegionJoin[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(T, X), Z]](
```

Comment:
Structurally, this could be placed in ShuffleRegionJoin, though it could be in either place in my opinion.